File: src/Commands/Normal/roleplay/baka.js
const { join } = require('path');
const fetch = require('node-fetch');
const { validate, parse } = require(join(__dirname, '../../../Functions/types/user'));

module.exports = {
    name: 'baka',
    aliases: [],
    description: 'It\'s not like I want you to use my command.. ~Baka!',
    ownerOnly: false,
    cooldown: 3000,
    userPermissions: ['SEND_MESSAGES'],
    clientPermissions: ['SEND_MESSAGES', 'EMBED_LINKS'],
    category: 'Roleplay',
    usage: '[user]',
    run: async (client, message, [ user ], Discord) => {
        try {
            const data = await fetch(`${process.env.API_URL}/api/roleplay?type=kiss`)
                .then(res => res.json())
                .catch(() => {});

            const embed = new Discord.MessageEmbed()
                .setColor('GREY')
                .setImage(data.url)
                .setFooter({ text: `Roleplay Commands | Made by Bear#3437 | ©️ ${new Date().getFullYear()} Tamako`, iconURL: client.user.displayAvatarURL({ dynamic: true }) });
            
            if (!user) return message.reply({ embeds: [embed] });
            if (!validate(user, message)) return message.reply('Please provide a valid user.');
            user = parse(user, message);
            
            if (user.id === client.user.id) return message.react('💢');
            if (user.id === message.author.id) return message.reply(`\\❌ No **${message.author.tag}**, you're not Baka!`);

            return message.reply({ embeds: [ embed.setDescription(`${user} B~baka!`) ] });

        } catch(err) {
            return message.reply({ content: `Let my developer know in the support server https://discord.gg/dDnmY56 or using \`${process.env.PREFIX}feedback\` command`, embeds: [ 
                new Discord.MessageEmbed()
                    .setColor('RED')
                    .setTitle('Error')
                    .setDescription(`\`${err}\``)
                    .setFooter({ text: `Error Occured | Made by Bear#3437 | ©️ ${new Date().getFullYear()} Tamako`, iconURL: client.user.displayAvatarURL({ dynamic: true }) })]
            });
        }
    }

};

/**
 * @INFO
 * Bot Coded by Bear#3437 | https://github.com/bearts
 * @INFO
 * Tamako Tech | https://tamako.tech/
 * @INFO
 */

File: pkg/commands/message_commands/modules/action/baka.go
package action

import (
	"context"

	"github.com/BearTS/Tamako/pkg/common/embed_maps/action_embeds"
	"github.com/andersfylling/disgord"
)

// BakaCommand struct
type BakaCommand struct{}

func (a BakaCommand) Name() string {
	return "baka"
}

func (a BakaCommand) Aliases() []string {
	return []string{}
}

func (a BakaCommand) Help() string {
	return "Sends a baka gif"
}

func (a BakaCommand) Description() string {
	return "Sends a baka gif"
}

func (a BakaCommand) Category() string {
	return "action"
}

func (a BakaCommand) Run(ctx context.Context, s disgord.Session, msg *disgord.Message) error {
	var embed *disgord.Embed

	if len(msg.Mentions) > 0 {
		embed = action_embeds.GetBakaEmbed(msg.Mentions[0])
	} else {
		embed = action_embeds.GetBakaEmbed(msg.Author)
	}

	// Send the reply
	if _, err := msg.Reply(context.Background(), s, embed); err != nil {
		return err
	}

	return nil
}
